docs(certificates): Rotate root cert (#181)
openservicemesh/osm-docs: content/docs/guides/certificates.md

For details and code where this is used see [osm-controller.go](https://github.com/openservicemesh/osm/blob/release-v0.9/cmd/osm-controller/osm-controller.go#L182-L183).

#### Rotate the Root Certificate

> The self-assigned root certificate, which is created via the Tresor package within OSM, will expire in a decade. To rotate the root cert, the following steps should be followed: 

Reviewer: ```suggestion
The self-signed root certificate, which is created via the Tresor package within OSM, will expire in a decade. To rotate the root cert, the following steps should be followed: 
```

Reviewer 2: Root certificate rotation applies to certificate providers other than Tresor as well. Could you clarify in the documentation whether the process to rotate the root cert is the same for all certificate providers, and additionally the steps for each of them if different? I think for Vault we do not create the osm-ca-bundle secret.

For details and code where this is used see [osm-controller.go](https://github.com/openservicemesh/osm/blob/release-v0.9/cmd/osm-controller/osm-controller.go#L182-L183).

### Rotate the Root Certificate (Tresor)

The self-signed root certificate, which is created via the Tresor package within OSM, will expire in a decade. To rotate the root cert, the following steps should be followed: 

1. Delete the `osm-ca-bundle` certificate in the osm namesapce
   ```console
   kubectl delete secret osm-ca-bundle -n osm-system
   ```
2. Restart the `osm-controller` pod(s) in the osm-namespace
   ```console
   kubectl delete pod <osm-controller-pod> -n osm-system
   ```

> 

Reviewer 2: We should document that all control plane components - osm-bootstrap, osm-controller, and osm-injector need to be restarted so they pick up the new root CA certificate similar to install.

For details and code where this is used see [osm-controller.go](https://github.com/openservicemesh/osm/blob/release-v0.9/cmd/osm-controller/osm-controller.go#L182-L183).

### Rotate the Root Certificate (Tresor)

The self-signed root certificate, which is created via the Tresor package within OSM, will expire in a decade. To rotate the root cert, the following steps should be followed: 

1. Delete the `osm-ca-bundle` certificate in the osm namesapce
   ```console
   kubectl delete secret osm-ca-bundle -n osm-system
   ```
2. Restart the control plane components in the osm-namespace
   ```console
   kubectl delete pod <osm-controller-pod> -n osm-system
   kubectl delete pod <osm-injector-pod> -n osm-system

>    kubectl delete pod <osm-bootstrap-pod> -n osm-system

Reviewer 2: Could you change the command to use `kubectl rollout restart deploy <deployment>` instead.

   ```

When the components gets re-deployed, you should be able to eventually see the new `osm-ca-bundle` secret in the osm-namespace:

```console
kubectl get secrets -n osm-system
```

```
NAME                           TYPE                                  DATA   AGE
osm-ca-bundle                  Opaque                                3      74m
```

The new expiration date can be found with the following command:

```console

> kubectl get secrets -n osm-system osm-ca-bundle -o json | jq -r '.data.expiration' | base64 -d

Reviewer 2: Would you mind using a variable to set the namespace:
`osm_namespace=<osm-namespace> # Replace <osm-namespace> with the namespace where OSM is installed`

and referencing `$osm_namespace` instead of `osm-system` here.

We do not want to mislead users into thinking it's always `osm-system`. 
This is the pattern followed in the demo docs at https://docs.openservicemesh.io/docs/demos/

1. Delete the `osm-ca-bundle` certificate in the osm namesapce
   ```console
   export osm_namespace=<osm-namespace> # Replace <osm-namespace> with the namespace where OSM is installed
   kubectl delete secret osm-ca-bundle -n $osm_namespace
   ```
2. Restart the control plane components in the osm-namespace
   ```console
   kubectl rollout restart deploy osm-controller -n $osm_namespace
   kubectl rollout restart deploy osm-injector -n $osm_namespace
   kubectl rollout restart deploy osm-bootstrap -n $osm_namespace
   ```

When the components gets re-deployed, you should be able to eventually see the new `osm-ca-bundle` secret in the osm-namespace:

```console
kubectl get secrets -n $osm_namespace
```

```
NAME                           TYPE                                  DATA   AGE
osm-ca-bundle                  Opaque                                3      74m
```

The new expiration date can be found with the following command:

```console
kubectl get secrets -n $osm_namespace osm-ca-bundle -o json | jq -r '.data.expiration' | base64 -d
```

#### Other certificate issuers

> For certificate providers other than Tresor, the process of rotating the root certificate will be different. For Hashicorp Vault, users will need to rotate the root certificate themselves outside of OSM. For cert-manager.io, the cert-manager.io API can be used to rotate the root certificate. 

Reviewer 2: ```suggestion
For certificate providers other than Tresor, the process of rotating the root certificate will be different. For Hashicorp Vault and cert-manager.io, users will need to rotate the root certificate themselves outside of OSM.
```
